src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -3010,9 +3052,24 @@ namespace winrt::TerminalApp::implementation
             co_return;
         }
 
+        auto pkgList = co_await _FindPackageAsync(args.MissingCommand());
+        for (int retries = 0; !pkgList && retries < 3; ++retries)
+        {
+            pkgList = co_await _FindPackageAsync(args.MissingCommand());
+        }
+
+        // no packages were found, nothing to suggest
+        if (pkgList.Size() == 0)
+        {
+            co_return;
+        }
+
         std::vector<hstring> suggestions;
-        suggestions.reserve(1);
-        suggestions.emplace_back(fmt::format(FMT_COMPILE(L"winget install {}"), args.MissingCommand()));
+        suggestions.reserve(pkgList.Size());
+        for (auto pkg : pkgList)
+        {
+            suggestions.emplace_back(fmt::format(FMT_COMPILE(L"winget install --id {}"), pkg.CatalogPackage().Id()));
```

Comment:
```suggestion
            suggestions.emplace_back(fmt::format(FMT_COMPILE(L"winget install -s winget --id {}"), pkg.CatalogPackage().Id()));
```
Without the source parameter, the id could collide with another package catalog.